### DISCUSSION TITLE
Explainer parameters for overall feature importance with uncalibrated binary node classification model

### DISCUSSION THREAD

**Discussion by lpigou** [pyg-team/pytorch_geometric#8494] [Q&A] [answered]
Hi!

I'm trying out different parameters for the Explainer class, but I'm getting confused what parameters to choose.

- My goal is to have the overall feature importance for the entire graph neural net, not just one node
- It's a binary node classification problem

Do I choose "phenomenon" or "model"?
Do I choose "common_attributes" or "attributes"?
Do I use the entire dataset in the index parameter: `explanation=explainer(.., index=np.arange(len(data)))`

I'm using the amazing GNNExplainer. The feature importances in the visualize_feature_importance function are just a summation of the node_mask. Don't we have to normalize them first for GNNExplainer?  @RexYing 
Also, are the feature importances additive: can I group up features somehow?

**Comment by rusty1s** [comment, marked as answer]
Which options you want to choose depends on what you want to explain and which output you want to receive. Phenomenon computes gradients based on the loss with respect to the target, while model computes gradients based on the loss with respect to the model output. Attributes/Common attributes depends on whether you want to receive feature-level importance or node/feature-level importance scores. Whether to explain per node or per graph is a good question, but also depends on the use-case. Usually, people are more interested in node-level explanations. However, graph-level explanations might be preferred if you want to find importance features across the whole dataset.
